nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,565 @@
+            };
+            "/staticfiles/".alias = "${cfg.api.staticRoot}/";
+          };
+        };
+        };
+      };
+
+      systemd.tmpfiles.rules = [
+        "d ${cfg.dataDir} 0755 funkwhale funkwhale -"
+        "d ${cfg.api.mediaRoot} 0755 funkwhale funkwhale -"
+        "d ${cfg.api.staticRoot} 0755 funkwhale funkwhale -"
+        "d ${cfg.musicDirectoryPath} 0755 funkwhale funkwhale -"
+      ];
+
+      systemd.services = 
+        let serviceConfig = {
+            User = "funkwhale";
+            WorkingDirectory = "${pkgs.funkwhale}";
+            EnvironmentFile =  "${funkwhaleEnvFile}";
+          };
+        in {
+        "funkwhale.target" = {
+          description = "Funkwhale";
+          wants = ["funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service"];
+        };
+        funkwhale-psql-init = {
+          description = "Funkwhale database preparation";
+          after = [ "redis.service" "postgresql.service" ];
+          wantedBy = [ "funkwhale-init.service" ];
+          before   = [ "funkwhale-init.service" ];
+          serviceConfig = {
+            User = "postgres";
+          };
+          script = ''
+            ${pkgs.postgresql}/bin/psql -d funkwhale -c 'CREATE EXTENSION IF NOT EXISTS "unaccent";'
+          '';
+        };
+        funkwhale-init = {
+          description = "Funkwhale initialization";
+          after = [ "funkwhale-psql-init.service" ];
+          wantedBy = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          before   = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          environment = funkwhaleEnv;
+          script = ''
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py migrate
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py collectstatic --no-input
+            if ! test -e ${cfg.dataDir}/createSuperUser.sh; then
+              echo "#!/bin/sh
+            
+              set -a
+              . ${funkwhaleEnvFile}
+              set +a
+              ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py createsuperuser" > ${cfg.dataDir}/createSuperUser.sh
+              chmod u+x ${cfg.dataDir}/createSuperUser.sh
+              chown -R funkwhale.funkwhale ${cfg.dataDir}
+            fi
+            if ! test -e ${cfg.dataDir}/config; then
+              mkdir -p ${cfg.dataDir}/config
+              ln -s ${funkwhaleEnvFile} ${cfg.dataDir}/config/.env
+              ln -s ${funkwhaleEnvFile} ${cfg.dataDir}/.env
+            fi
+          '';
+        };
+
+        funkwhale-server = {
+          description = "Funkwhale application server";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = "${pythonEnv}/bin/daphne -b ${cfg.apiIp} -p ${toString cfg.apiPort} config.asgi:application --proxy-headers";
```

Comment:
Don't use `script` when it's just a single command, use `serviceConfig.ExecStart` instead, same for the other services.

Reply:
fixed